src/test/java/com/mongodb/kafka/connect/util/config/ConfigSoftValidatorTest.java

```diff
@@ -108,6 +109,27 @@ void logIncompatibleProperties() {
     assertEquals(expectedMessages, actualMessages);
   }
 
+  @Test
+  void logIncompatiblePropertiesWithNullValues() {
+    Set<IncompatiblePropertiesPair> incompatibleConfigs =
+        Stream.of(IncompatiblePropertiesPair.latterIgnored("a", "", "c", null))
+            .collect(Collectors.toSet());
+    Map<String, String> props = new HashMap<>();
+    props.put("a", "valueA");
+    props.put("c", null);
+    props.put(topicOverridePropertyName("t", "c"), "valueC");
+    Set<String> expectedMessages =
+        Stream.of(latterIgnoredMsg("a", topicOverridePropertyName("t", "c"))).collect(toSet());
+    Set<String> actualMessages = new HashSet<>();
+    ConfigSoftValidator.logIncompatibleProperties(incompatibleConfigs, props, actualMessages::add);
+    assertEquals(expectedMessages, actualMessages);
+
+    props.put(topicOverridePropertyName("t", "c"), null);
+    actualMessages = new HashSet<>();
+    ConfigSoftValidator.logIncompatibleProperties(incompatibleConfigs, props, actualMessages::add);
+    assertTrue(actualMessages.isEmpty());
+  }
```

Comment:
Instead of adding a new test, I propose to modify the existing one. That way the amount of test code remains roughly unchanged (the modified test still fails when run against the current `master` due to each of the new `null` values). See https://github.com/stIncMale/mongo-kafka/commit/935410d412b471f3871c13b7295805b2e51f5261.

Reply:
Thanks - incorporated your change